pkg/controllers/nodepool/readiness/controller.go

```diff
@@ -57,16 +59,25 @@ func (c *Controller) Reconcile(ctx context.Context, nodePool *v1.NodePool) (reco
 	if len(supportedNC) == 0 {
 		logger.Fatal("no supported node classes found for the cloud provider")
 	}
-	nodeClass, err := c.getNodeClass(ctx, nodePool, supportedNC)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	if nodeClass == nil {
-		nodePool.StatusConditions().SetFalse(v1.ConditionTypeNodeClassReady, "NodeClassNotFound", "NodeClass not found on cluster")
-	} else if !nodeClass.GetDeletionTimestamp().IsZero() {
-		nodePool.StatusConditions().SetFalse(v1.ConditionTypeNodeClassReady, "NodeClassTerminating", "NodeClass is Terminating")
+	var err error
```

Comment:
I don't see this error getting referenced elsewhere. What's the need for it being at the top of all of these calls?